src/v/tristate.h

```diff
@@ -51,6 +51,16 @@ class tristate {
         return !is_disabled() && get_optional().has_value();
     }
 
+    /// \brief Checks if the tristate is in the "Disabled or "Set" states.
+    /// We often use "Not set" as the default state of a tristate, and
+    /// it's useful to check if any explicit changes were made.
+    ///
+    /// \return true if the tristate is in the "Disabled" or "Set" state
+    ///         false otherwise
+    constexpr bool is_engaged() const {
```

Comment:
nit: I don't feel strongly about it, but if the reason for going with `is_engaged` over `is_set` is the comments around this method, and you want to actually name this `is_set`, consider updating the comments to "Value" and "No value" respectively.

Reply:
Sort of. When I first saw this tristate I read the top level doc string which outlined the 3 states ("Disabled", "Set", "Not set") and that made sense for me. Naming it `is_set` was going against the wiring in my brain.